plugins/woocommerce/includes/admin/settings/class-wc-bis-admin-settings.php

```diff
@@ -0,0 +1,348 @@
+<?php
+/**
+ * WC_BIS_Settings class
+ *
+ * @package  WooCommerce Back In Stock Notifications
+ * @since    1.0.0
+ */
+
+use Automattic\WooCommerce\Internal\BackInStockNotifications;
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+if ( ! class_exists( 'WC_BIS_Settings' ) ) :
+
+	/**
+	 * WooCommerce Back In Stock Notifications Settings.
+	 *
+	 * @class    WC_BIS_Settings
+	 * @version  1.3.0
+	 */
+	class WC_BIS_Settings extends WC_Settings_Page {
+
+		/**
+		 * Constructor.
+		 */
+		public function __construct() {
+
+			$this->id    = 'bis_settings';
+			$this->label = __( 'Stock Notifications', 'woocommerce' );
+
+			// Add subtab to WC > Settings > Products.
+			add_filter( 'woocommerce_get_sections_products', array( $this, 'add_bis_section_to_product_settings' ), 100, 1 );
+
+			// Customer stock notifications subtab settings.
+			add_filter( 'woocommerce_get_settings_products', array( $this, 'add_customer_stock_notifications_settings' ), 100, 2 );
+
+		}
+
+		public function add_bis_section_to_product_settings( array $sections ): array {
+			// Add bis_settings section to products tab after Inventory section.
+			$inventory_index = array_search( 'inventory', array_keys( $sections ) );
+			if ( $inventory_index !== false ) {
+				$sections = array_slice( $sections, 0, $inventory_index + 1, true ) +
+					array( 'bis_settings' => __( 'Customer stock notifications', 'woocommerce' ) ) +
+					array_slice( $sections, $inventory_index + 1, null, true );
+			} else {
+				$sections['bis_settings'] = __( 'Customer stock notifications', 'woocommerce' );
+			}
+	
+			return $sections;
+		}
+
+
+		/**
+		 * Handler for 'woocommerce_get_settings_products', adds the settings related to the Back In Stock Notifications.
+		 *
+		 * @param array  $settings Original settings configuration array.
+		 * @param string $section_id Settings section identifier.
+		 * @return array New settings configuration array.
+		 *
+		 * @internal For exclusive usage of WooCommerce core, backwards compatibility not guaranteed.
+		 */
+		public function add_customer_stock_notifications_settings( array $settings, string $section_id ): array {
+			if ( 'bis_settings' === $section_id ) {
+				$title_item = array(
+					'title' => __( 'Customer stock notifications', 'woocommerce' ),
+					'type'  => 'title',
+				);
+
+				if ( ! BackInStockNotifications::is_enabled() ) {
```

Comment:
I don't know how we can end up here. Considering this controller won’t function if the feature is disabled, right?

Reply:
You're right, that's an atavism from when the settings were handled differently. Removed in [3d439dd](https://github.com/woocommerce/woocommerce/pull/55787/commits/3d439ddc32efc74722c6f17fe87d1c2311c6467b).